components/WithdrawLiquidity/WithdrawLiquidityController.tsx

```diff
@@ -1,26 +1,269 @@
+          const _token0 = new ethers.Contract(token0Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
+          const _token1 = new ethers.Contract(token1Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
+          const symbol0 = await _token0.symbol()
+          const symbol1 = await _token1.symbol()
+          setToken0({address: token0Addr, symbol: symbol0, contract: _token0})
+          setToken1({address: token1Addr, symbol: symbol1, contract: _token1})
+        } else {
+          let _depPool = new ethers.Contract(address, DepPool.abi, provider)
+          setdepPool(_depPool)
+          const token0Addr = await _depPool.token0()
+          const token1Addr = await _depPool.token1()
+          const _token0 = new ethers.Contract(token0Addr, IERC20Metadata.abi, provider)
+          const _token1 = new ethers.Contract(token1Addr, IERC20Metadata.abi, provider)
+          const symbol0 = await _token0.symbol()
+          const symbol1 = await _token1.symbol()
+          setToken0({address: token0Addr, symbol: symbol0, contract: _token0})
+          setToken1({address: token1Addr, symbol: symbol1, contract: _token1})
+        }
+      } else {
+        console.log('Please connect wallet')
+      }
+    }
+    fetchContract()
+  }, [provider])
+
+  function pretty(num: number) {
+    return ethers.utils
+      .formatEther(num.toString())
+      .toString()
+      ?.match(/^-?\d+(?:\.\d{0,2})?/)![0]
+  }
+
+  useEffect(() => {
+    async function fetchData() {
+      if (!depPool) {
+        return
+      }
+      const liqBal = await depPool.balanceOf(accountInfo?.address)
+      setTotalLiquidityAmt(liqBal.toString())
+      setLiquidityAmt(parseFloat(pretty(liqBal.toString())))
+
+      const uniPair = await depPool.getUniPair()
+      if (!provider) {
+        return
+      }
+
+      const uniPairContract = new ethers.Contract(uniPair, IUniswapV2Pair.abi, provider)
+      const reserves = await uniPairContract.getReserves()
+      const _uniPrice = BigNumber.from(reserves.reserve1).mul(BigNumber.from(10).pow(18)).div(reserves.reserve0)
+      setUniPrice(_uniPrice.toString())
+      const liqBalNum = BigNumber.from(liqBal.toString())
+
+      if (liqBalNum.gt(constants.Zero) && _uniPrice.gt(constants.Zero)) {
+        setLiqInTokB(
+          parseFloat(
+            pretty(
+              sqrt(_uniPrice.mul(BigNumber.from(10).pow(18)))
+                .mul(liqBalNum)
+                .div(BigNumber.from(10).pow(18))
+                .mul(2)
+                .toString()
+            )
+          )
+        )
+      } else {
+        setLiqInTokB(0)
+      }
+    }
+    fetchData()
+  }, [depPool])
+
+  function sqrt(y: any) {
+    let z
+    if (y.gt(3)) {
+      z = y
+      let x = y.div(2).add(1)
+      while (x.lt(z)) {
+        z = x
+        x = y.div(x).add(x).div(2)
+      }
+    } else if (!y.isZero()) {
+      z = BigNumber.from(1)
+    }
+    return z
+  }
```

Comment:
might be better to extract this function into the `utils` folder for reusability.

Reply:
Ok

Reply:
Done